### PR TITLE
support set sub field config

### PR DESCRIPTION
**Pull request by qiuzuhui**
support case like : add config for compress:src2 when already exists config for compress:src1

### CODE REVIEW

**Comment by qiuzuhui** [comment]
just show some idea. :)
what i want is add sub task config when already have some sub task.

File: index.js [hunk 1]
```diff
@@ -25,16 +25,27 @@ var GruntfileEditor = module.exports = function (gruntfileContent) {
  * Insert a configuration section inside the `Gruntfile.js` initConfig() block
  * @param  {String} name   - Key name of the configuration block
  * @param  {String} config - Configuration content code as a string.
+ * @param  {String} options - additional options
+ *                  {Boolean} keyPath: support child field.
```
Comment (SBoudrias) [inline index.js]: `@param {Boolean} options.keypath - support child field`

File: index.js [hunk 2]
```diff
@@ -25,16 +25,27 @@ var GruntfileEditor = module.exports = function (gruntfileContent) {
...
   name = _.isString(name) && name.trim() || false;
   config = _.isString(config) && config.trim() || false;
   assert(name, 'You must provide a task name');
   assert(config, 'You must provide a task configuration body as a String');
-  this.gruntfile.callExpression('grunt.initConfig').arguments.at(0)
-    .key(name).value(config);
+  var gConfig = this.gruntfile.callExpression('grunt.initConfig').arguments.at(0)
+  if(options && options.keyPath){
+    // support set config for compress:src2 when already exists config for compress:src1
+    name.split('.').forEach(function(k){
```
Comment (SBoudrias) [inline index.js]: Why split on the `.`?

**Comment by SBoudrias** [comment]
To add a new feature, you'll need to add unit test validating it's working.

**Comment by SBoudrias** [comment]
Hey @qiuzuhui I'd love to merge this in, would you have time to address the review comments?

**Comment by qiuzuhui** [comment]
Hi SBoudrias, sorry for update so long time :) , updated jsdoc and added test for it :).

**Comment by SBoudrias** [comment]
I manually merge this on master as I cleaned up your commits and simplified the code: https://github.com/SBoudrias/gruntfile-editor/commit/a6d0bf9c24e595fb321bf07402a695fe29f53ea9

**Pull request closed without merging**